### PR TITLE
feat: [M3-7644] - Add VPC IPv4 address and range to Linode IP Address Table

### PR DESCRIPTION
**Pull request by hana-linode**
## Description 📝
Add support for displaying VPC IPv4 addresses and ranges in the Linode Details -> Network tab -> "IP Addresses" table

## Changes  🔄
List any change relevant to the reviewer.
- Add Support for VPC IPv4 addresses and ranges to the Linode IP Address table
  - If there is a VPC interface with 1:1 NAT, hide the Public IPv4 IP address row
- Improve mobile text display of the Type column

## Preview 📷
| Before  | After   |
| ------- | ------- |
| ip address table before | image |

| Mobile before  | Mobile after   |
| ------- | ------- |
| mobile before | image |

### Prerequisites
(How to setup test environment)
- Have the `vpc-extra-beta` tag on your account and a VPC with at least one Subnet
- Assign a Linode to the Subnet and check both the checkboxes (Auto-assign a VPC IPv4 address & Assign a public IPv4 address)
- To add `ip_ranges`, check out https://github.com/linode/manager/pull/10089

### Verification steps 
(How to verify changes)
- Go to the Linode Details -> Network tab of the assigned Linode and check the IP Addresses table
- You should see IP Address rows for `IPv4 – VPC`, `VPC IPv4 – NAT`, and `IPv4 – VPC – Range` for each range
- You should _not_ see an IP address row for the type `IPv4 - Public`
```
yarn test LinodeIPAddressRow
```

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Linodes/LinodesDetail/LinodeNetworking/LinodeIPAddresses.tsx [hunk 1]
```diff
@@ -286,6 +296,42 @@ export interface IPDisplay {
...
+  }
+
+  if (configInterfaceWithVPC.ipv4?.nat_1_1) {
+    ipDisplay.push({
+      address: configInterfaceWithVPC.ipv4.nat_1_1,
+      type: 'VPC IPv4 – NAT',
+    });
+  }
+
+  if (configInterfaceWithVPC.ip_ranges) {
```
Comment (jaalah-akamai) [inline packages/manager/src/features/Linodes/LinodesDetail/LinodeNetworking/LinodeIPAddresses.tsx:323]: Nit: we can deconstruct these: `const { ipv4, ip_ranges } = configInterfaceWithVPC;`

File: packages/manager/src/features/Linodes/LinodesDetail/LinodeNetworking/LinodeIPAddresses.tsx [hunk 2]
```diff
@@ -121,6 +123,14 @@ export const LinodeIPAddresses = (props: LinodeIPAddressesProps) => {
   const renderIPTable = () => {
     const ipDisplay = ipResponseToDisplayRows(ips);
 
+    if (configInterfaceWithVPC) {
+      ipDisplay.push(
+        ...(vpcConfigInterfaceToDisplayRows(
+          configInterfaceWithVPC
+        ) as IPDisplay[])
```
Comment (jaalah-akamai) [inline packages/manager/src/features/Linodes/LinodesDetail/LinodeNetworking/LinodeIPAddresses.tsx:130]: We should look into a way not to cast this. I don't want to hold this up, but we should circle back if you can't find a solution sooner.

**Review by jaalah-akamai — APPROVED** [review]
✅  Otherwise, I'm able to add linode to vpc and see it's ranges in the network table

File: packages/manager/src/features/Linodes/LinodesDetail/LinodeNetworking/LinodeIPAddresses.tsx (hunk 2 above)
Comment (hana-linode) [reply packages/manager/src/features/Linodes/LinodesDetail/LinodeNetworking/LinodeIPAddresses.tsx]: Good point. Refactored to avoid casting

**Review by jaalah-akamai — APPROVED** [review]
✅ Confirmed IP Address table is populating with VPC IPs
✅ Public IPv4 address is hidden

**Review by dwiley-akamai — APPROVED** [review]
`IPv4 – VPC`, `VPC IPv4 – NAT`, and `IPv4 – VPC – Range` observed in "IP Addresses" table where expected ✅
`IPv4 - Public` row not displayed in 1:1 NAT scenario ✅
Code review ✅
Mobile view ✅

**Pull request merged**